<a href="https://colab.research.google.com/github/Thobeee/Hamoye-Project/blob/main/Hamoye_Assessment1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Hamoye Assessment 1

In [1]:
#Importing the important libraries

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
#Loading the dataset
url ='https://raw.githubusercontent.com/WalePhenomenon/climate_change/master/fuel_ferc1.csv'
df = pd.read_csv(url)
df.head()

record_id  ...  fuel_cost_per_mmbtu
0   f1_fuel_1994_12_1_0_7  ...                1.121
1  f1_fuel_1994_12_1_0_10  ...                1.120
2   f1_fuel_1994_12_2_0_1  ...                1.650
3   f1_fuel_1994_12_2_0_7  ...                1.970
4  f1_fuel_1994_12_2_0_10  ...                2.570

[5 rows x 11 columns]

In [3]:
df.shape

(29523, 11)

In [4]:
df['record_id'].nunique()

29523

## Questions and Answers

### 1. Which of the following fuel type code has the lowest average fuel cost per unit burned

In [5]:
df.groupby('fuel_type_code_pudl')['fuel_cost_per_unit_burned'].mean().sort_values()

fuel_type_code_pudl
gas          13.659397
other        18.253856
waste        19.518122
coal         67.421830
oil         168.877086
nuclear    4955.157002
Name: fuel_cost_per_unit_burned, dtype: float64

**Gas** has the lowest average fuel cost per unit

### 2. What is the standard deviation and 75th percentile of the measure of energy per unit(Fuel_mmbtu_per_unit) in two decimal place

In [6]:
standard_deviation = np.std(df['fuel_mmbtu_per_unit'])
percentile_75 = np.percentile(df['fuel_mmbtu_per_unit'],75) 
print(f'{standard_deviation:.2f}')
print(f'{percentile_75:.2f}')

10.60
17.01


The standard deviation of the measure of energy per unit(Fuel_mmbtu_per_unit) in two decimal place is **10.60**

The 75th percentile of the measure of energy per unit(Fuel_mmbtu_per_unit) in two decimal place is **17.01**

### 3. What is the skewness and kurtosis for the fuel quantity burned in two decimal places?

In [7]:
#Import kurtosis and skew from scipy module
from scipy.stats import kurtosis, skew
skewness = skew(df['fuel_qty_burned'])
kurt = kurtosis(df['fuel_qty_burned'])
print(f'{skewness:.2f}')
print(f'{kurt:.2f}')

15.85
651.26


The skewness for the fuel quantity burned in two decimal place is **15.85**

The kurtosis for the fuel quantity burned in two decimal place is **651.37**

### 4. Which feature has missing values and what is the total number of missig value and percentage of the missing rows as a factor of the total number of rows in three decimal places

In [9]:
null_feature = df.columns[df.isnull().any()].tolist()
for value in null_feature:
   total = df[value].isnull().sum()
   percentage = df[value].isnull().sum()/df.shape[0] * 100

   print(f'Feature:{value}, Total:{total}, Percent:{percentage:.3f}')

Feature:fuel_unit, Total:180, Percent:0.610


The feature with null values: **fuel_unit**

The total number of null values: **180**

The percentage of the total: **0.610**

### 5. Which features has the second and third lowest correlation with the Fuel Cost Per Unit Burned

In [44]:
corr.loc['fuel_cost_per_unit_burned'].sort_values(ascending=False)

fuel_cost_per_unit_burned       1.000000
report_year                     0.013599
fuel_cost_per_unit_delivered    0.011007
fuel_cost_per_mmbtu            -0.000437
fuel_mmbtu_per_unit            -0.010034
fuel_qty_burned                -0.018535
utility_id_ferc1               -0.037863
Name: fuel_cost_per_unit_burned, dtype: float64

The features that have the second and third lowest correlation with fuel Cost Per Unit Burned are **fuel_qty_burned** and **fuel_mmbtu_per_unit** respectively

### 6. For the fuel type coal, what is thepercentage change in the fuel cost per unit in 1998 compared to 1994

In [28]:
coal = df[df['fuel_type_code_pudl'] == 'coal']
percent_coal_burned_in_1998 = coal[coal['report_year'] == 1998]['fuel_cost_per_unit_burned'].sum() 
percent_coal_burned_in_1994 = coal[coal['report_year'] == 1994]['fuel_cost_per_unit_burned'].sum()
percent_change = (percent_coal_burned_in_1998 - percent_coal_burned_in_1994)/percent_coal_burned_in_1998
percent = percent_change * 100
print(percent)

-25.893298748163968


The percentage change in fuel cost burned in 1998 compared to 1994 for coal is **-26%**

### 7. Which year has the highest average fuel cost per unit delivered

In [35]:
new_df = df[['report_year', 'fuel_cost_per_unit_delivered']]
new_df.groupby('report_year')['fuel_cost_per_unit_delivered'].mean().sort_values(ascending = False)

report_year
1997    11140.197239
1996     9196.705948
2000      985.362877
2009      652.694163
2018      499.269966
2015      326.535511
1998      287.154420
2014      192.737183
2013      172.307591
2004      139.524275
2016      103.901761
2010       91.862105
1994       63.636060
2012       60.994502
2001       60.050396
2011       59.774667
2008       58.588197
2003       55.663493
2002       47.594361
2017       46.196861
2007       43.325023
2005       41.438184
2006       38.657484
1995       32.735269
1999       25.551627
Name: fuel_cost_per_unit_delivered, dtype: float64

The year with the highest average fuel cost per unit delivered is **1997**